### PR TITLE
Visualize transaction execution timings

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

1. there's no easy way to see how's going with regards with scheduling for **both replay stage and banking stage**.

2. ryoqun never try to merge things, piling half-baked prs.

#### Summary of Changes

Try vizsualize it to solve the 1:

image

for 2., ...meh. ;)

#### bare minimum docs for your fun:

Summary:

transaction timings are sent to influxdb first. then we fetch it via influxdb's ajax api from codepen-hosted front-end (d3.js/plot.js) directly inside browser.

alternatively, you can use binaries from https://github.com/ryoqun/solana/tree/banking-simulator-wip-with-unified-scheduler-from-tokyo-5-grand-integration in similar way

A: data recording setup

A.1. cherry-pick (and backport if needed) this
A.2. run local influxdb and initial configuration
A.3. run solana-ledger-tool with XXXXX

B: data viewing setup 

B.1. run lcp (a) with paired ssh tunnels (b)
B.2. open codepen (fork it and edit the query source code as you like)

Detailed steps:

A.1.

```
$ git
```

A.2.

```
$ sudo docker run --detach --publish 8086:8086 influxdb:1.7
```

**note that it's quick to fill up the docker image's data dir with sheer amount of individual transaction execution timings data!**

```
$ influx -host localhost
...
> CREATE DATABASE "transaction-timings"
```

A.3.

like this:

```
$ (set -x && SOLANA_METRICS_CONFIG="host=http://localhost:8086,db=transaction-timings,u=foo,p=bar" SOLANA_METRICS_MAX_POINTS_PER_SECOND=1000000 SOLANA_TRANSACTION_TIMINGS=1 EXECUTING_THREAD_COUNT=1 $(type -ap work/solana/target/release/solana-ledger-tool) --ledger mainnet-beta-bench-2022-09/ verify --accounts-db-skip-shrink --skip-poh-verify) &> ledger-tool-capitalization.log.$(date '+%Yy%mm%dd%Hh%Mm%Ss%Nns')
```

**NOTE: There's `assert!()`, but never spam solana's official influxdb with normal `SOLANA_METRICS_CONFIG`!**

B.1a.

install https://github.com/garmeeh/local-cors-proxy for your fav way on your local machine for your web browser. then run it like this (replace the ip; for convenience, this ip is publicly accessible for trouble-shooting):

```
$ ./node_modules/.bin/lcp --proxyUrl http://34.168.201.175:8088/ --port 8011
```

note: this is needed due to https://github.com/influxdata/influxdb/issues/1268

B.1b:

given:

```
$ sudo docker ps
CONTAINER ID   IMAGE          COMMAND                  CREATED        STATUS        PORTS                                       NAMES
ba9f0217c6ed   influxdb:1.7   "/entrypoint.sh infl…"   3 months ago   Up 3 months   0.0.0.0:8086->8086/tcp, :::8086->8086/tcp   admiring_mahavira
```

```
ssh -L localhost:8086:localhost:8086 you@your-host
```

B.2.

open and interact with the ui (i.e. source code): https://codepen.io/ryoqun/full/yLvmPrr

spin-off from: #27666 
context: #23548

### CODE REVIEW

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -171,6 +171,95 @@ use {
...
+                }
+            })
+            .unwrap();
+
+        Some(sender)
+    });
+
+    if let Some(sender) = maybe_sender {
+        sender
+            .send_buffered(TransactionTimings {
```
Comment (ryoqun) [inline runtime/src/bank.rs:248]: this is needed for super light-weight off-loading; needs to upstream to the crossbeam. :)

**Comment by ryoqun** [comment]
I've improved a bit to show tx dep arrows and tps/cumulative tx count:

onset of new slot processing (thread pool is so much saturated with flood of schedulable txes!):

image


closing part of another slot (the tx is executed concurrently while dep is resolved as much as tightly):

image

**Comment by overclock-validator** [comment]
Have you played with banking thread count?

**Comment by ryoqun** [comment]
> Have you played with banking thread count?

@overclock-validator: hey, super delayed reply. but making a progress. yep. my scheduler can run as banking mode and it can scale linearly around 20-40 cores:

image

**Pull request closed without merging**